scalameta/trees/shared/src/main/scala/scala/meta/internal/trees/InternalTrees.scala

```diff
@@ -72,6 +72,10 @@ trait InternalTree extends Product {
     }
   }
 
+  def withDialect(dialect: Dialect): Tree = {
+    this.privateCopy(origin = Origin.Generated(dialect))
```

Comment:
even if we wanted to do this, two points:
- the new `Origin` class is not really generated, it's a `DialectOnly`
- this particular method should only modify the origin if it's `Origin.None`

Reply:
One other problem I see with this implementation is that it always returns the `Tree` type instead of sticking to the sub-type.